### PR TITLE
Launcher Redesign

### PR DESCRIPTION
**Pull request by devendrn**
This pull request introduces several changes to the design aspect of the launcher. The new redesign, which is based on the official Minecraft Launcher, follows a dark color scheme and uses a different layout featuring a sidebar for navigation between settings, news, and play screens. The goal of this redesign is **not to make a Minecraft Launcher clone** but to make the launcher look more modern. The new redesign also fixes a couple of small issues previously present (eg. Scroll overflow in dev settings).

All newly added/modified assets were made by me.

The background art is loaded from `mcpelauncher/background_art`. Only `.jpg` and `.png` images are chosen, and the image is chosen at random when the Play screen is created. If no image is found, `noise.png` is used.

Note: I used the auto-format feature of Qt Creator during development, so all qml files were reformatted.

I don't expect this PR to get merged because of the sheer amount of changes I've made. But if you are thinking otherwise let me know what more changes I have to make/revert.

### CODE REVIEW

**Comment by ChristopherHX** [comment]
I did some testing and my qt6.7 built segfaults when doing the following

- Press Download and Play 
- Wait for the progress bar to change it's state to show an absolute progress
- Press Open Settings while Download is in progress

The problem is `GoogleApkDownloadTask` is destructed while navigating away from Play screen, so this of the async process is invalid memory.
Can we keep them alive? e.g. don't unload them...

> I don't expect this PR to get merged because of the sheer amount of changes I've made. But if you are thinking otherwise let me know what more changes I have to make/revert.

I'm pretty optimistic that I would accept this

**Comment by ChristopherHX** [comment]
GameLogWindow scrolls to the beginning automatically even when no log is added

**Comment by ChristopherHX** [comment]
Qt6 only standardButtons must be migrated to buttons, could be an older problem. No qt6 build has integrated updates configured to be enabled..
```
        onUpdateCheck: function (available) {
            updateCheckerConnectorSettings.enabled = false
            if (available) {
                updateInfo.text = qsTr("An Update of the Launcher is available for download") + "<br/>" + (columnlayout.updateUrl.length !== 0 ? qsTr("You can download the new Update here: %1").arg(columnlayout.updateUrl) + "<br/>" : "") + qsTr("Do you want to update now?")
                updateInfo.buttons = StandardButton.Yes | StandardButton.No
            } else {
                updateInfo.buttons = StandardButton.Ok
                updateInfo.text = qsTr("Your installed Launcher Version %1 (build %2) seems uptodate").arg(LAUNCHER_VERSION_NAME || '').arg(LAUNCHER_VERSION_CODE)
            }
            updateInfo.open()
        }
```

**Comment by devendrn** [comment]
> GameLogWindow scrolls to the beginning automatically even when no log is added

I can't seem to understand this issue. Can you tell me how I can replicate it?

**Comment by devendrn** [comment]
> Qt6 only standardButtons must be migrated to buttons, could be an older problem. No qt6 build has integrated updates configured to be enabled..

Shouldn't that be `MessageDialog` button in Qt6?
```qml
        onUpdateCheck: function (available) {
            updateCheckerConnectorSettings.enabled = false
            if (available) {
                updateInfo.text = qsTr("An Update of the Launcher is available for download") + "<br/>" + (columnlayout.updateUrl.length !== 0 ? qsTr("You can download the new Update here: %1").arg(columnlayout.updateUrl) + "<br/>" : "") + qsTr("Do you want to update now?")
                updateInfo.buttons = MessageDialog.Yes | MessageDialog.No
            } else {
                updateInfo.text = qsTr("Your installed Launcher Version %1 (build %2) seems uptodate").arg(LAUNCHER_VERSION_NAME || '').arg(LAUNCHER_VERSION_CODE)
                updateInfo.buttons = MessageDialog.Ok
            }
            updateInfo.open()
        }
```

**Comment by ChristopherHX** [comment]
> > GameLogWindow scrolls to the beginning automatically even when no log is added
> 
> I can't seem to understand this issue. Can you tell me how I can replicate it?

Seems to be an problem with my cached copy of qt6.7, possible a beta that is not caused by your change

**Comment by ChristopherHX** [comment]
> > Qt6 only standardButtons must be migrated to buttons, could be an older problem. No qt6 build has integrated updates configured to be enabled..
> 
> Shouldn't that be `MessageDialog` button in Qt6?

You are right, it's in the qt6 docu. I migrated the old code to qt6 by fixing errors, not by reading the docu.

File: qml/LauncherMain.qml [hunk 1]
```diff
@@ -27,103 +27,142 @@ LauncherBase {
...
+                    playVerChannel.playApi = null
+                    playVerChannel.playApi = playApiInstance
                 } else {
                     if (needsDownload()) {
                         if (googleLoginHelper.account === null) {
-                            launcherSettingsWindow.show();
-                            return;
+                            launcherSettingsWindow.show()
+                            return
                         }
```
Comment (ChristopherHX) [inline qml/LauncherMain.qml]: Please remove, seems like the button was already disabled when the account is null and `launcherSettingsWindow` has been deleted by your change

**Review by ChristopherHX — COMMENTED** [review]
I like your redesign.
I will look next week again into your PR, large diff
In my opinion it's only some bug squashing required.
Some thoughts from my side what I may  or may not further change after merge
- I may add a Import Wallaper button for png jpg or the zip file from minecraft.net
- I may remove the current "Minecraft" Style Logo, which might be better be avoided due to it's similarity

**Review by ChristopherHX — APPROVED** [review]
Thank you for redesigning, I'm merging the qt6 part now.
While cherry picking the qt6 ui to qt5 the sidebar was unusable for me, not shure what is wrong.

**Pull request merged**